### PR TITLE
[TS-819]<fix>: windows negative timestamp error.

### PR DESCRIPTION
**Pull request by afwerar**

### CODE REVIEW

File: src/kit/shell/src/shellEngine.c [hunk 1]
```diff
@@ -455,7 +455,28 @@ static char* formatTimestamp(char* buf, int64_t val, int precision) {
...
+    b.dwLowDateTime=c.LowPart;  
+    b.dwHighDateTime=c.HighPart;  
+    FileTimeToLocalFileTime(&b,&b);
+    FileTimeToSystemTime(&b,&a);
+    int pos = sprintf(buf,"%02d-%02d-%02d %02d:%02d:%02d", a.wYear, a.wMonth,a.wDay, a.wHour, a.wMinute, a.wSecond);   
+    if (precision == TSDB_TIME_PRECISION_NANO) {
+      sprintf(buf + pos, ".%09d", ms);
+    } else if (precision == TSDB_TIME_PRECISION_MICRO) {
+      sprintf(buf + pos, ".%06d", ms);
+    } else {
```
Comment (sangshuduo) [inline src/kit/shell/src/shellEngine.c:475]:
```
suggestion:
if (nanosec)
   process_nanosecond
else if (microsec)
   process_microsec
else if (millisec)
   process_millisec
else
   prompt error
```

File: src/kit/shell/src/shellEngine.c [hunk 2]
```diff
@@ -455,7 +455,28 @@ static char* formatTimestamp(char* buf, int64_t val, int precision) {
   */
 
 #ifdef WINDOWS
-  if (tt < 0) tt = 0;
+  if (tt < 0) {
+    SYSTEMTIME a={1970,1,5,1,0,0,0,0}; // support 1601-01-01 00:00:00
```
Comment (sangshuduo) [inline src/kit/shell/src/shellEngine.c]: confused why comment is 1601 but variable is 1970

Comment (sangshuduo) [reply src/kit/shell/src/shellEngine.c]: SYSTEMTIME struct support 1601-01-01. set 1970 to compatible with Epoch time

**Review by sangshuduo — CHANGES_REQUESTED** [review]

**Review by sangshuduo — APPROVED** [review]
LGTM

**Review by stephenkgu — APPROVED** [review]

**Pull request merged**